### DISCUSSION TITLE
Introducing Custom Speaker Identification and Labeling

### DISCUSSION THREAD

**Discussion by KentonParton** [deepgram/community#534] [Feature Requests]
Hi All,

Deepgrams speech-to-text service with diarization has been invaluable with its remarkable performance and accuracy. However, one feature that would greatly enhance its usefulness for us is the ability to recognize and label specific speakers.

Currently, your system assigns a generic "speaker" label followed by a number. This requires us to manually listen to the recordings, identify each speaker, and then update the speech-to-text output with the names of the speakers.

It would be highly beneficial if we could create a database of speaker profiles (speaker fingerprints). This way, when your system encounters a voice matching one of these profiles, it could automatically apply the custom label (name of speaker).

If Deepgram is open to exploring this enhancement, I'd love to collaborate.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @KentonParton, feedback received and I've logged it internally. We've heard interest in this type of feature. For transparency, we're putting more of our focus on cutting-edge ASR models, as well as audio intelligence features (such as summarization, sentiment analysis, intent recognition, topic detection), and text-to-speech. I hope that you'll continue to find value in Deepgram and enjoy our new features over time!

**Reply by KentonParton** [reply]
Thanks for the transparency @jkroll-deepgram!

**Comment by everuribe** [comment]
@jkroll-deepgram Just wanted to +1 that outputting speaker embeddings / "fingerprints" and exposing an API for historical speaker comparison / identification would be highly beneficial in terms of more effective diarization across recordings

**Comment by bmaluijb** [comment]
This would be a great feature to have! I would use it.

**Comment by enrique-gm** [comment]
+1

**Comment by pprobst** [comment]
+1

**Comment by avibathula** [comment]
Definitely a +1. Thanks @KentonParton for logging this request.

I presume there is some sort of quantifiable/probabilistic voice-signature that the transcription engine is calculating besides the spoken content.

Within a single transcription, given that DeepGram is able to identify and distinguish portions of the spoken content that maps to the same voice-signature and are marking them appropriately. And I would expect voice-signature to be something like an embedding vector with the ability to measure the distance between two such signatures and deduce that they are close enough to be marked as same speaker.

As DeepGram deals with high scale, I wouldn't expect them to store names of people provided on a per account basis and embed those names into transcription response JSON. It might also have privacy implications too, that DeepGram might prefer to not get into.

Instead, if only DeepGram can just stash the speaker voice-signature-embedding and its confidence-rate into the response JSON, we on the clientside can convert that into appropriate names.

Just sharing a few thoughts - let's discuss.

@team-deepgram, @jkroll-deepgram - please please pretty please consider putting some resources onto this request.

**Comment by venusbhatia** [comment]
+1, it is actually so well needed! Thanks

**Comment by KentonParton** [comment]
Hi all, (@venusbhatia @avibathula @pprobst @enrique-gm @bmaluijb @everuribe) I've implemented a speaker finger-printing API for a side-project of mine. This is the flow:

1. Start by generating a Deepgram diarized transcript.
2. Loop through transcript  finding 1 or multiple 5 second clips for each speaker where the Deepgram confidence is highest and is above 90%
3. I then generate the finger-print which is an embedding.
4. Search vector store for a matching embedding, if there is no embedding, I insert the embedding with the speakers name
5. If there are embeddings in the vector store with a confidence above 80%, I return them.

I generally create multiple fingerprints for one speaker from different videos to get the most reliable results.

Is this what everyone is after? If so, I can make it available on my website for purchase. If your use-cases differs, let me know.

**Reply by avibathula** [reply]
It has been on my back burner, and I was hoping to get to it next month.

I would urge you to consider open sourcing the functionality and providing it as a return-gift to the community and gain street creds. 

---
Pretty please ...

![pretty-please-puss-and-boots-eyes-9k98qaajfjbpt7ni-ezgif com-webp-to-gif-converter](https://github.com/deepgram/community/assets/5514875/58575302-db41-413d-9912-6bb4f4c8f274)
Source : https://gifdb.com/gif/pretty-please-puss-and-boots-eyes-9k98qaajfjbpt7ni.html

**Comment by gergomiklos** [comment]
+1. Also, the current live diarization is unusable.

**Comment by holtaila** [comment]
+

**Comment by jpvajda** [comment]
Related: https://github.com/orgs/deepgram/discussions/561

**Comment by fripokoff** [comment]
Hello we need this

**Comment by bradtem** [comment]
I will add desire for this feature.     I would love to be able to pre-train the diarization.    One could either pre-feed a library of audio of each speaker, or one could have deepgram process the samples and output fingerprint profiles for each speaker which could be saved and re-loaded when doing a new audio stream.      I have noticed that the deepgram diaraization does make mistakes, and often identifies the same speaker as 2 or more distinct speakers.  However, if possible it would be nice to tell it, "Speaker 3 and 5 are the same" so that it combines the fingerprints, or just maps them to the same person in future.

And while I am on the feature list (this may deserve another post) it would be very interesting to get "music detection" which would tell if it is hearing speaking or singing.    (Not identification of what the melody is, that's for other tools.)    This would allow breaking up recordings that have both speaking and singing into their components, and might even improve recognition in the sung sections as currently it doesn't do very well on those.
